src/it/resources/com/google/checkstyle/test/chapter2filebasic/rule231filetab/InputWhitespaceCharacters.java

```diff
@@ -90,26 +92,29 @@ private InputWhitespaceCharacters() {
     // a line
   }
 
-  /** test local variables */
+  /** test local variables. */
   private void localVariables() {
     int abc = 0;
-    int ABC = 0;
+    int pqr = 0;
 
     final int cde = 0;
-    final int CDE = 0;
+    final int xyz = 0;
 
-    for (int k = 0; k < 1; k++) {
+    for (int testing = 0; testing < 1; testing++) {
       String innerBlockVariable = "";
     }
-    for (int I = 0; I < 1; I++) {
-      String InnerBlockVariable = "";
+    for (int testing = 0; testing < 1; testing++) {
+      String innerblockvariable = "";
     }
   }
 
-  /** test method pattern */
+  /** test method pattern. */
   void ALL_UPPERCASE_METHOD() {}
+  // 2 violations above:
+  //  'Abbreviation in name .* must contain no more than '1' consecutive capital letters.'
+  //  'Method name 'ALL_UPPERCASE_METHOD' must match pattern.'
```

Comment:
why we cannot avoid this violations ?

https://checkstyle.org/google_style.html#a2.3.1
https://checkstyle.org/checks/whitespace/filetabcharacter.html#Violation_Messages
https://github.com/search?q=path%3Asrc%2Fmain%2Fresources%2Fcom%2Fpuppycrawl%2Ftools%2Fcheckstyle%2Fchecks%2Fwhitespace+path%3A**%2Fmessages*.properties+repo%3Acheckstyle%2Fcheckstyle+%22containsTab%22&type=code
https://github.com/search?q=path%3Asrc%2Fmain%2Fresources%2Fcom%2Fpuppycrawl%2Ftools%2Fcheckstyle%2Fchecks%2Fwhitespace+path%3A**%2Fmessages*.properties+repo%3Acheckstyle%2Fcheckstyle+%22file.containsTab%22&type=code

Reply:
this Test is about tab characters, all other violation should not be present.

Reply:
I thought this method was to show that method name can contain all uppercase characters so I kept this violation.

Reply:
If you want to keep it to show violation, move it to inputs related to Check that gives violation.

This chapter Input should have violations only about Tab characters.